Example/Tests/CapturedImagesStackViewTests.swift

```diff
@@ -19,31 +19,38 @@ final class CapturedImagesStackViewTests: XCTestCase {
     }
     
     func testCaptureStackWhenNoImages() {
-        capturedImagesStackView.updateStackStatus(to: .empty)
+        capturedImagesStackView.replaceStackImages(with: [])
         
         XCTAssertTrue(capturedImagesStackView.isHidden,
                       "capturedImagesStackView should be hidden when there are no images")
         
     }
     
     func testCaptureStackWhenOneImageCaptured() {
-        capturedImagesStackView.updateStackStatus(to: .filled(count: 1,
-                                                              lastImage: loadImage(withName: "invoice.jpg")!))
+        let images = [loadImage(withName: "invoice.jpg")!]
+        
+        capturedImagesStackView.replaceStackImages(with: images)
         
         XCTAssertFalse(capturedImagesStackView.isHidden,
                       "capturedImagesStackView should not be hidden when it is filled")
         XCTAssertTrue(capturedImagesStackView.thumbnailStackBackgroundView.isHidden,
                       "thumbnailStackBackgroundView should be hidden when there is only 1 image")
+        XCTAssertEqual(capturedImagesStackView.thumbnailButton.image(for: .normal), images[0],
```

Comment:
I disagree a little bit here. You don't have to test the image in all methods - it's redundant. And now you have three assertions.

I would argue that you have 3 tests:
* having one image -> you test that the stack view is not hidden
* having two images -> you test that the background view is not hidden
* having at least one image -> you test that the image view shows the thumbnail of the top most image.